src/main/kotlin/com/bridgecrew/services/CliService.kt

```diff
@@ -1,43 +1,70 @@
+import java.nio.charset.Charset
+import javax.swing.SwingUtilities
 
 private val LOG = logger<CliService>()
 
 @Service
 class CliService {
+    var checkovPath: String = ""
+    var checkovVersion: String = ""
 
-    fun run(command: String, env: Array<String>? = null, dir: File? = null): String {
-        val commandToPrint = replaceApiToken(command)
+    fun run(
+        commands: ArrayList<String>,
+        project: Project,
+        function: (output: String, exitCode: Int, project: Project) -> Unit,
+    ) {
+        val commandToPrint = commands.joinToString(" ")
         LOG.info("Running command: $commandToPrint")
-        val checkovProcess = Runtime.getRuntime().exec(command, env, dir)
-        val result = checkovProcess.inputStream.bufferedReader().use { it.readText() }
-        val error = checkovProcess.errorStream.bufferedReader().use { it.readText() }
-        val checkovExitCode = checkovProcess.waitFor()
-
-        if (checkovExitCode != 0) {
-            LOG.info("Failed to run cli $commandToPrint")
-            LOG.info(error)
-            throw Exception("Failed to run cli command")
-        }
-
-        LOG.info("Finished running command: $commandToPrint")
-
-        return result
+        val generalCommandLine = GeneralCommandLine(commands)
+        generalCommandLine.charset = Charset.forName("UTF-8")
+        generalCommandLine.setWorkDirectory(project.getBasePath())/////////////??????????
 
+        val processHandler: ProcessHandler = OSProcessHandler(generalCommandLine)
+        val myBackgroundable =
+            BackgroundableTask(project, "running cli command $commandToPrint", processHandler, function )
+        if (SwingUtilities.isEventDispatchThread()) {
+            ProgressManager.getInstance().run(myBackgroundable)
+        } else {
+            ApplicationManager.getApplication().invokeLater {
+                ProgressManager.getInstance().run(myBackgroundable)
+            }
+        }
     }
 
-    private fun replaceApiToken (command: String): String{
-        val apiToknIndex = command.indexOf("--bc-api-key")
-        return if (apiToknIndex >= 0){
-            val firstPos: Int = apiToknIndex + "--bc-api-key".length
-            val lastPos: Int = command.indexOf("--repo-id", firstPos)
-            command.substring(0, firstPos).toString() + " **-**-**-** " + command.substring(lastPos)
-        } else{
-            command
+    private class BackgroundableTask(
+        project: Project,
+        title: String,
+        val processHandler: ProcessHandler,
+        val function: (output: String, exitCode: Int, project: Project) -> Unit,
+    ) :
+        Task.Backgroundable(project, title,true) {
+        override fun run(indicator: ProgressIndicator) {
+            indicator.isIndeterminate = false
+            val output =
+                ScriptRunnerUtil.getProcessOutput(processHandler,
+                    ScriptRunnerUtil.STDOUT_OR_STDERR_OUTPUT_KEY_FILTER,
+                    720000000)
```

Comment:
whats that number? perhaps it's better to use a variable that will indicate what it is

Reply:
A random number, changed it to be the default value of GeneralCommand object